indico/modules/events/abstracts/legacy.py

```diff
@@ -0,0 +1,506 @@
+        return AbstractDescriptionFieldProxy(self.event)
+
+    def _notifyModification(self):
+        pass
+
+    def hasField(self, field_id):
+        if field_id == 'content':
+            return True
+
+        legacy = self.event.contribution_fields.filter(ContributionField.legacy_id == field_id)
+        if legacy:
+            return True
+        else:
+            return AbstractFieldWrapper.wrap(
+                self.event.contribution_fields.filter(ContributionField.id == int(field_id)).count()) > 0
+
+    def getFieldById(self, field_id):
+        if field_id == 'content':
+            field = self.description_field
+        else:
+            field = self.event.contribution_fields.filter(ContributionField.legacy_id == field_id).first()
+            if not field:
+                field = self.event.contribution_fields.filter(ContributionField.id == int(field_id)).first()
+        return AbstractFieldWrapper.wrap(field) if field else None
+
+    def getFields(self):
+        fields = [AbstractFieldWrapper.wrap(field) for field in self.event.contribution_fields]
+        fields.append(AbstractFieldWrapper.wrap(AbstractDescriptionFieldProxy(self.event)))
+        return fields
+
+    def getActiveFields(self):
+        fields = [AbstractFieldWrapper.wrap(field) for field in self.event.contribution_fields if field.is_active]
+        fields.append(AbstractFieldWrapper.wrap(AbstractDescriptionFieldProxy(self.event)))
+        return fields
+
+    def hasActiveField(self, field_id):
+        if field_id == 'content':
+            return self.description_field.is_active
+        return self.getFieldById(field_id).isActive() if self.getFieldById(field_id) else False
+
+    def hasAnyActiveField(self):
+        return (self.event.contribution_fields.filter(ContributionField.is_active).count() > 0 or
+                self.description_field.is_active)
+
+    def enableField(self, field_id):
+        field = self.getFieldById(field_id)
+        if field:
+            field.is_active = True
+
+    def disableField(self, field_id):
+        field = self.getFieldById(field_id)
+        if field:
+            field.is_active = False
+
+
+class AbstractDescriptionValue(object):
+    """Simulates a ``ContributionFieldValue``."""
+
+    def __init__(self, abstract):
+        self.abstract = abstract
+
+    @property
+    def contribution_field(self):
+        return AbstractDescriptionFieldProxy(self.abstract)
+
+    @property
+    def data(self):
+        return self.abstract.description
+
+    @data.setter
+    def data(self, description):
+        self.abstract.description = description
+
+
+class AbstractLegacyMixin(object):
+    """Implements legacy interface of ZODB ``Abstract`` object."""
+
+    def _get_field_value(self, field_id):
+        if field_id == 'content':
+            return AbstractDescriptionValue(self.as_new)
+        else:
+            field = ContributionField.find_first(legacy_id=field_id, event_new=self.event)
```

Comment:
Actually, we have `Contribution.get_field_value(field_id)`. Maybe we should add the same method to `Abstract`?